frontend/javascripts/oxalis/view/action-bar/tracing_actions_view.js

```diff
@@ -172,10 +180,27 @@ class TracingActionsView extends React.PureComponent<Props, State> {
     isShareModalOpen: false,
     isMergeModalOpen: false,
     isUserScriptsModalOpen: false,
+    isReopenAllowed: false,
   };
 
   modalWrapper: ?HTMLDivElement = null;
 
+  componentDidUpdate = () => {
+    const localStorageEntry = UserLocalStorage.getItem("lastFinishedTask");
+    if (this.props.task && localStorageEntry) {
+      const { finishedTime } = JSON.parse(localStorageEntry);
+      if (Date.now() - finishedTime < features().taskReopenAllowedFrontend) {
+        this.setState({
+          isReopenAllowed: true,
+        });
+        setTimeout(
+          () => this.setState({ isReopenAllowed: false }),
```

Comment:
Two things:
- after the timeout was hit, I'd do `UserLocalStorage.removeItem("lastFinishedTask");` to clean up.
- The `setTimeout` is a bit "dirty" right now. Ideally, the timeout should be cleared when the component is unmounted (via `clearTimeout`). Otherwise, the callback can be called when the component is not mounted anymore which will fail due to the `this.setState` code. Also, repeated calls of `componentDidUpdate` can schedule multiple timeouts which are then executed in parallel. You can solve this issue à la:
```js
class TracingActionsView extends React.PureComponent<Props, State> {
    reopenTimeout: ?number

    componentDidUpdate() = () => {
      ...
      if (this.reopenTimeout != null) {
        clearTimeout(this.reopenTimeout);
      }
      this.reopenTimeout = setTimeout(...);
    }

    componentWillUnmount() {
      if (this.reopenTimeout != null) {
        clearTimeout(this.reopenTimeout);
      }
    }
}
```

(not tested)